### ISSUE TITLE
Question about the plone_patterns_settings view

### ISSUE THREAD

**Issue by ale-rt** [plone/Products.CMFPlone#4064]
## Question

I have some doubts about the `plone_patterns_settings` view.

The `data-base-url` is defined as `self.context.absolute_url()`, see:

https://github.com/plone/Products.CMFPlone/blob/26c659aaad12ce59376d3d79ca12b2d1128fa38a/Products/CMFPlone/patterns/view.py#L26

This caused some javascript issues when using a view with a custom traverser (something like: `@@custom-view/foo/bar`).
What I experienced was that some JS ajax calls were done to my context url rather than my "traversed conxtet" URL:

I solved this issue by setting `data-base-url` to `context_state.current_base_url()`.

Sample Code:
```python 
class CustomPatternsSettingsView(BasePatternsSettingsView):

    def __call__(self):
        context_state = getMultiAdapter(
            (self.context, self.request), name="plone_context_state"
        )
        result = super().__call__()
        if isinstance(self.request["PARENTS"][0], EditItemForm):
            result["data-base-url"] = context_state.current_base_url()
        return result
```

Should we rename the `data-base-url` to `data-context-url`?
Is it right that in the case of a view like `@@custom-view/foo/bar` the `data-base-url` should be the context URL?

### What version of Plone/ Addons I am using:

Plone 6.0-latest, but this also happens on Plone 6.1.

**Comment by petschki** [comment]
I would assume, that the `base-url` (comming from the earlier used `<base>` tag) has the URL of the base content context. 

Changeing its name will break some `mockup` implementations notably the registry view.

Maybe it would be better to use the `@@plone_context_state/current_base_url` instead (https://github.com/plone/plone.app.layout/blob/master/plone/app/layout/globals/context.py#L43)

**Comment by ale-rt** [comment]
> Maybe it would be better to use the `@@plone_context_state/current_base_url` as value instead (https://github.com/plone/plone.app.layout/blob/master/plone/app/layout/globals/context.py#L43)

Yes, that's the one I am using (I updated my sample code that was badly anonymized by AI).

**Comment by jensens** [comment]
Verified: patterns/view.py uses self.context.absolute_url() for data-base-url instead of current_base_url(). Needs decision on correct default.
